### DISCUSSION TITLE
Text Preset Button | Vivo Type

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1253] [Proposals]
### Goal

Criar um token (button-text-preset) para aplicar no componente Button.

### Context

No Mística, usamos o peso Medium para as Labels dos botões, mas não temos temos Medium na nova tipografia da Vivo. Aplicando o peso Bold, ficaria com uma visualização muito pesada, com isso, queremos aplicar o peso Regular no componente Button.

### Impact

Deixar a tipografia do componente Button mais fluída e consistente. 

### Discussion Questions

É possível criar essa token de tipografia para utilizar na nova skin da Vivo? 

### Do you have a visual proposal?

https://www.figma.com/file/c78Ch1q62Y7DjbFWeFU2Wz/Button-update-font-size?type=design&node-id=0%3A1&t=Y5e4TnpYDV9YBRcv-1

**Comment by aweell** [comment]
Hi @annefranca!

I have a question prior to talk about creating a specific token for the button.

What happen with the rest of the places where the medium weights are used? For example: Text3 medium is used in the Button but also in the Tabs component. Text2 medium is used in the Tooltip title, Chip and Buttons (small variant in mobile and link in desktop).

If all the cases where text4, text3, text2 and text1 medium are going to regular? If that's the case instead of creating multiple specific tokens for the button component, the configuration can be more global, if not, wee need to evaluate case by case what we need to cover.

Are currently planned cases to use Bold weights in the hub project?

**Reply by annefranca** [reply]
@aweell 

Aqui você pode visualizar a Type Vivo "old" e "new" e entender suas diferenças de peso. 

Figma: https://www.figma.com/file/aNHyvXe1mqjaelE8XWVQD5/%F0%9F%94%B9-Text-presets-definition?type=design&node-id=1374%3A1294&t=1piGEtnNLzGYdHQT-1

**Reply by aweell** [reply]
Hi @annefranca,

The Figma link provided seems broken, or the file was moved.

**Reply by annefranca** [reply]
@aweell vê se agora dá certo https://www.figma.com/file/aNHyvXe1mqjaelE8XWVQD5/%F0%9F%94%B9-Text-presets-definition?type=design&node-id=0%3A1&t=aZl5IPgLfMtKS4AB-1

**Reply by BrunoBiagioniNeto** [reply]
@aweell , você viu o Figma? Lá nós documentamos que as marcas atuais não mudam. Acredito que faltou explicar isso:

Goal

Criar um token (button-text-preset) para aplicar no componente Button e ter a possibilidade de alterar somente no tema vivo um novo peso para a fonte do botão. As demais marcas não mudam o preset.

Marcas Globais - button text-preset > preset medium (não mudam)
Marca Vivo New - button text-preset > preset regular (muda só pra vivo)

![image](https://github.com/Telefonica/mistica-design/assets/123083886/6f4f03f8-b332-4c5d-ab60-e54ff367125a)

**Reply by aweell** [reply]
Hey @annefranca & @BrunoBiagioniNeto 

Now I was able to see the document, sorry was my fault (incorrect account logged). What I'm seeing here is two different approaches we can apply to solve this need:


1. In case that **all the components that are currently using medium weights change to regular** we can then map, when using Vivo type, that [the weights 400 and 500 are regular](https://github.com/Telefonica/mistica-web/blob/master/doc/fonts.md). The bold would be still be available for atomic builds since we're planning to update the support in presets from 1-4 for the four weights: light, regular, medium and bold.
2. In case **not all the components will change their weight to regular**. In that case is where we need to identify which ones change to create specific tokens like the one proposed in this discussion.

So, the list of the components currently using medium weights in the presets from 1-4 are:

* Button (small variants included)
* Tabs
* Chips
* Tooltip

**Reply by aweell** [reply]
@annefranca  & @BrunoBiagioniNeto 

Here an [example of how the **option 1** would work without creating any component token](https://mistica-web.vercel.app/playroom/preview/#?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhlQJ4AOE2GA2saaYSZ6UQIYBbGPAwEQANQCWAN2YAVRjHEAaDrwwAbKQHMAFlVHjpcjIoYwMAGV0HV63gCcYOgK6b+jo5NkKl9nl5hKClXQW8TPwsAjQwAIwhNKAjfMyUMACFEqBjOAF81QPwHTgFhbwAlCASqCFzebX1DMRAqmuYbJvrOZzcPLxa2iFru0mDQ8MHq4eYAWRgQsNH47Mrp2szs7oL1AF0AbnVyagwKAGspIgA5IUssWj0YYQA6c8ub4UOidU0YGgAzCBEKgAYSB-10X1IFAA7lIqGA9BhkFRHi83tdbphuLwwPwKJZxLMIDIpNRPOJ4CVSIDgWCiBCdCwAIxQjRxZz8M5s0h4gktADyACZKdSMLTQeDdCyeZwOTAubK+YSQPIYL9aVI8aKijSgZKGdL7qyxfLFSVlS0Mh5XDrYhL6YyZabOdyLfiVZE7RoHVKmfcAAyy0hmt08PLEdTOKiuRwkZAlAA8ODFiYACs5STAYfIIIkKDhEwB6DMwLM5vOaAuR3UYRPUOi-HDAAAGNYSUDoxQZ+oAtP8hFJNHRRL3+AwGL9exQ6NQnioMAASYD0JisX2GnS7Z5lGAFDAAcgAEuqZH8tfwDwuDwBBRxSfiaK+nfiUacwe--L4R75EAACEr9vwYCWNwgEDoIQ4jkuK6MBA676o6ujbrueRfIBDa-KIRAQI4giPuhfYwjAtjNAAzAGQbEBQjhgKImgQHimjIMuq7wRujLbo0Bh5GghQ-gBfYDiB3bgYOw6iKxcEIXSfoobcaHEBh9BYRgOF4QRSlESRTSiAALJRXw0XRWiMY+LGwWuHHIc8vTuJ4vH8cQgnAkBIlgUJ4nQVJVmIXJO4KYRrmYSIam4fhmhBVQvbEaRogAKyGdRtH0WZzE+exfmbtu4xhI5xACWJ7laa5EFQZJlmZbJ2UBcIik9sFKmhepEVRTFOkGKIADsSWUClplMRZbEyQanHPAkST5UQP7EC2eTFiFKa1omWQAB4YAw-BQCERA6FgwAABwBnkS2xKQYqkImADKVDAWcpybSB+16QdJ0XZwiYgo+DGuFQ72kFQ8K-Fg4ggnor46JYqKWByr5QLQzCkqYGwElDjzivqZDg7tMAUOI-0YLAFBgPeDCA0CIM+KY5iWEQMALBQCPxJYlzkpovzwwxTHDvjtakEWp1nZdN13Q9wEwPtzIAGxvXzGiJmqq1UGROCdAYxaK8rgtC5dmuE7jZy1AwV1SAAXhLwAGXkGCxU0lPcX9IDazrpDyOjACOrhavdHIQDCJCAutABWYQMIzJIfg8lgeKbXZQBAOgE5dRaa87QsKzASsqxULj2Y4GuZ1rSd1nrROGxAxtmxbVtpzrbuWJ73vxI4fsBxAweh+HZ6OFHWj8LHhMJ8XBdK7XsQZ1nODzIsgj4ND6IXJiwjYFg9zGL44gYAA-C0yBXbcGD4hgdn9Ggm+iK4RCwBCdNQPNKeFyrw+lwbRsm+bz3HTbHVUJTuWCOIMeZ164YEbmAH2Ld-YYw7oIMOGAI492hn3Ae8dE5y14CPKgQCMGaxVlkJImCn7oI+i-c4b8q6f2trbAwlMJo5CdsXV2HsvbgObq3aBGAQ6wK7pHJBMc45D2IcnVOBNiwi3AUtUR4j7oUEetXV62CPoAEkiDaDpmLJ6wBpayxdhgjIv1ahEAzFIfCjguxAlLBQCg+10DYDwMAPIJ0bxgHJkQYs+iqCGOMaYugii9EGIsfeHxpwIqaHgUY5wVibGYFwPgRxfiNDONcYwusRYPFeKCZ4XxKSVoBKIFdGA5AoBZPCZY6xKAYn2PiUkqQQJ3F5IKUUrJCTLrpKBI0oExSzEhO+qUyJ5TbGxIcTo3RvAalAhyWkhphTOnNJyW0ogAARCGkdAm4wGZUuJTiXG1LcVMzxQJlk40cC0usCyjmQx7hQUJfT1nRLsVs05pBxm-lGcnc5KyTnzLyTYIg901lRIqQ84ZOAXn1IOUQX5ZwEnFhUWomAML5D8DiHjIRnACS-BcQsFRsBVr7UhlQEWVAYAJhALdOIOLM6qAwAGNAP43nhLBis-aBJCW3WJaS8llLVriDpSk8l5T2Bot4DiBlnBiVK28EiuIGBmTLFGVqIEohExKPIDeCcEBLhUGEMCHOfRPBkESLhSmYBYzOFkgxRwm8BaFDFRgHYwrOCirtRK5o4hpUYBFCAW1dqMCKqIMq1VQIMiOH4JcPVedDWWpNWamAFrcLWpwD6hlDq7XOrFa6qVyKMBkXlbo-1gbyD6IoBG-oUbjXiFNY4c1kpLWJuTW81Nbzdj0t0QLGsozEwVC2rUgA4i3VwDA1K3BNXoKQDBew6AHQwTe19+DuCoBIR8rgJbiDlQwx1KrVGXEsLI8W+0FEpMut2kIEAFlHs4DIZdq6QDrovaQc1sBHA2OAFjQpZwFgLg8HEdUSj4a8QeQmR148wbjvvRoYCgMzz7URO+hYrbfWcCkFAfa37f133A7wINRB9rYbVlQIglwdAADEhwcwQ76p5Z1QNDrlcB+WRYaNUdIHy+j7b6Ndp7WevJ4Gr2aBXZTL14HH0fhfW+8Bn6+4-s0H++1mygOIcujRzDnBIOyAtrBiTGH6OxGQ6h5F6GKOIew7h8g+HCO7VI+zeDKnmMaBo56lTxYmPgdY3a9jdrOOnvPTpjAfGBPiFzSAYTcan1ic0x+qAX6DMyf-fJpzynfO8DU9B4AEWbNJaQyh4AaHYtGd9SZ4AeHSIWZI2RjLimMB2d4A53NmXnNjoYHZtzYqPNvNhduumMKiwnr7dOxRYjboSPeoNu6td3Ht2dsWQWaBvxoHQD+b1ZK0RwEQMSUk5JHAgDyEAA)

> **Note** 
> Text from 1-4 will be changed to introduce bold as a prop configuration.

**Reply by annefranca** [reply]
@aweell teríamos como alterar apenas text-preset 2 Medium e o text-preset 3 Medium para Regular? Faço essa pergunta porque são essas duas tipografias/pesos que estão impactando ocomponente Buttom e outros componentes. Vou listar eles aqui:

- Button
- Chip
- Tag
- Text Link
- Badge

Outra dúvida: se no futuro a Vivo aderir o peso Medium da família tipográfica novamente, seria fácil "readequarmos" novamente?

**Reply by aweell** [reply]
Hi @annefranca,

> teríamos como alterar apenas text-preset 2 Medium e o text-preset 3 Medium para Regular? 

If there's any case where `text-preset-1/medium` and `text-preset-4/medium` are used in a component and should go to `bold` we should explore more how to handle the change. If in all components where any of the presets from 1-4 change their `weight` from `medium` to `regular`... I suggest to go with the [option 1](https://github.com/Telefonica/mistica-design/discussions/1253#discussioncomment-5976309).

##### Cases where `text-preset-1/medium` and `text-preset-4/medium` are used:

I was digging further to try to find all the cases:

* **Title 1**: The title 1 component is using `text-preset-1/medium` (I see [the proposal](https://www.figma.com/file/JkA7OefLqga6XyfiqLq1dY/Vivo---Lib-Intermedi%C3%A1ria?type=design&node-id=1524%3A104918&t=xYJXHIFMGGYOeGE7-1) is to change to regular so it shouldn't be a problem?)
* **Stepper**: the step indicator is using  `text-preset-1/medium`



--------------------------

> se no futuro a Vivo aderir o peso Medium da família tipográfica novamente, seria fácil "readequarmos" novamente?

Would't be a problem, when the `medium` weight is available, the font configuration can be updated to stop using `regular` and start using `medium`.

**Reply by annefranca** [reply]
@aweell validamos internamente com o Rodrigo Pires. Vamos seguir com o text-preset 2 Medium e o text-preset 3 Medium alterando para o peso Regular. 

Quais seriam os próximos passos?

**Reply by annefranca** [reply]
@aweell podemos dar continuidade da forma que comentei acima?

**Comment by yceballost** [comment]
Vamos a crear un token de peso tipográfico para gestionar los pesos medium que se están usando en los componentes actuales

**Reply by yceballost** [reply]
https://github.com/Telefonica/mistica-design/issues/1285
